src/lib/types.ts

```diff
@@ -50,6 +50,12 @@ export type PostgrestSingleResponse<T> =
   | PostgrestResponseFailure
 export type PostgrestMaybeSingleResponse<T> = PostgrestSingleResponse<T | null>
 
+const delay = (ms: number) => {
+  return new Promise((resolve) => setTimeout(() => resolve(), ms))
+}
+
+const RetryableErrorCodes = new Set([503, 504, 522, 523, 524])
```

Comment:
I think 504, 522, 523, 524 are fine to retry on.

503 means that PostgreSQL is unreachable for postgrest, retrying in that case could be handled in another PR since PostgREST provides a Retry-After header. Later on 503 will also signal when postgrest's pool is overloaded.